### PR TITLE
azure-cosmos: Fixing unit test failure in extended CI tests

### PR DESCRIPTION
**Pull request by FabianMeiswinkel**

### CODE REVIEW

File: sdk/cosmos/azure-cosmos/src/test/java/com/azure/cosmos/implementation/directconnectivity/BarrierRequestHelperTest.java [hunk 1]
```diff
@@ -178,7 +178,7 @@ public void barrierWithAadAuthorizationTokenProviderType() throws URISyntaxExcep
                 false,
                 null,
                 null,
-                new CosmosClientTelemetryConfig(),
+                new CosmosClientTelemetryConfig().sendClientTelemetryToService(false),
```
Comment (xinlian12) [inline sdk/cosmos/azure-cosmos/src/test/java/com/azure/cosmos/implementation/directconnectivity/BarrierRequestHelperTest.java:181]: should not this be false by default?

**Review by xinlian12 — APPROVED** [review]
LGTM, thanks

File: sdk/cosmos/azure-cosmos/src/test/java/com/azure/cosmos/implementation/directconnectivity/BarrierRequestHelperTest.java (hunk 1 above)
Comment (FabianMeiswinkel) [reply sdk/cosmos/azure-cosmos/src/test/java/com/azure/cosmos/implementation/directconnectivity/BarrierRequestHelperTest.java:181]: when coming thourgh public API = yes. But some tests create RxDocumentSeviceClient manually - and there it is expected that the effecive flag is passed down. CosmosClinetBuilder is doing it when coming through public API.

**Pull request merged**